embedded/src/main/java/org/wildfly/core/embedded/Configuration.java

```diff
@@ -0,0 +1,421 @@
+ */
+public interface Configuration {
+
+    /**
+     * Hints which logger provider should be used.
+     */
+    enum LoggerHint {
+        DEFAULT(null, null),
+        // Don't set the system package for the JBoss Log Manager. This causes some loggers to be set on the log context
+        // from the logging subsystem as opposed to the log context created by the user.
+        JBOSS_LOG_MANAGER("jboss", null),
+        LOG4J("log4j", "org.apache.log4j"),
+        LOG4J2("log4j2", "org.apache.logging.log4j"),
+        LOGBACK("slf4j", "org.slf4j"),
+        JUL("jdk", null);
+
+        private final String providerCode;
+        private final String systemPackage;
+
+        LoggerHint(final String providerCode, final String systemPackage) {
+            this.providerCode = providerCode;
+            this.systemPackage = systemPackage;
+        }
+
+        /**
+         * The provider code for the JBoss Logging provider.
+         *
+         * @return the provider code
+         */
+        public String getProviderCode() {
+            return providerCode;
+        }
+    }
+
+    /**
+     * The path to the servers directory.
+     *
+     * @return the servers directory
+     */
+    Path getJBossHome();
+
+    /**
+     * The module loader to use.
+     *
+     * @return the module loader
+     */
+    ModuleLoader getModuleLoader();
+
+    /**
+     * An array of boot arguments.
+     *
+     * @return the boot arguments or an empty array
+     */
+    String[] getCommandArguments();
+
+
+    /**
+     * A builder for creating the {@linkplain Configuration configuration}
+     */
+    @SuppressWarnings({"WeakerAccess", "unused", "UnusedReturnValue"})
+    class Builder {
+        private static final String SYSPROP_KEY_CLASS_PATH = "java.class.path";
+        private static final String SYSPROP_KEY_EMBEDDED_LOGGING = "org.wildfly.logging.embedded";
+        private static final String SYSPROP_KEY_JBOSS_MODULES_DIR = "jboss.modules.dir";
+        private static final String SYSPROP_KEY_LOGGING_PROVIDER = "org.jboss.logging.provider";
+        private static final String SYSPROP_KEY_MODULE_PATH = "module.path";
+        private static final String SYSPROP_KEY_SYSTEM_MODULES = "jboss.modules.system.pkgs";
+
+        private static final String JBOSS_MODULES_DIR_NAME = "modules";
+
+        private static final AtomicBoolean MODULE_LOADER_CONFIGURED = new AtomicBoolean(false);
+
+        private final Path jbossHome;
+        private final List<String> cmdArgs;
+        private final List<String> systemPackages;
+        private LoggerHint loggerHint;
+        private ModuleLoader moduleLoader;
+        private String modulePath;
+        private boolean embeddedLoggingHint;
+
+        /**
+         * Creates a new builder for the configuration.
+         *
+         * @param jbossHome the servers home directory
+         */
+        public Builder(final File jbossHome) {
```

Comment:
should the Builder constructors be private as they seem to be hidden behind the `of` methods?